api/src/main/scala/cryptonite/ProductsService.scala

```diff
@@ -28,14 +30,17 @@ class ProductsService(
       gdaxTickers <- EitherT(gdaxGateway.read())
       krakenTickers <- EitherT(krakenGateway.read())
       bitfinexTickers <- EitherT(bitfinexGateway.read())
+      bitstampTickers <- EitherT(bitstampGateway.read())
       gdaxAmounts <- gdaxTickers.traverse(x => EitherT(portfolioService.get(x.product.base.index.code)))
       krakenAmounts <- krakenTickers.traverse(x => EitherT(portfolioService.get(x.product.base.index.code)))
       bitfinexAmounts <- bitfinexTickers.traverse(x => EitherT(portfolioService.get(x.product.base.index.code)))
+      bitstampAmounts <- bitstampTickers.traverse(x => EitherT(portfolioService.get(x.product.base.index.code)))
     } yield {
       val gdaxBooks = gdaxTickers.zip(gdaxAmounts).map { case (t, a) => createBook(Exchange.GDAX, t, a) }
       val krakenBooks = krakenTickers.zip(krakenAmounts).map { case (t, a) => createBook(Exchange.Kraken, t, a) }
       val bitfinexBooks = bitfinexTickers.zip(bitfinexAmounts).map { case (t, a) => createBook(Exchange.Bitfinex, t, a) }
-      List.concat(gdaxBooks, krakenBooks, bitfinexBooks)
+      val bitstampBooks = bitstampTickers.zip(bitstampAmounts).map { case (t, a) => createBook(Exchange.Bitstamp, t, a) }
+      List.concat(krakenBooks, bitfinexBooks, bitstampBooks)
```

Comment:
Direi che è abbastanza evidente l'interfaccia comune che dovrebbero avere i gateway, non trovi?

Reply:
A breve con filtering e sorting sarò obbligato a fare refactoring di questa for comprehension 😃